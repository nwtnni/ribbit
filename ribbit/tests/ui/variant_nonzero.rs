#[ribbit::pack(size = 16, nonzero)]
#[derive(Copy, Clone)]
enum Foo {
    #[ribbit(size = 8)]
    Bar(u8) = 1,
    #[ribbit(size = 8)]
    Baz(u8) = 0,
}

fn main() {}
