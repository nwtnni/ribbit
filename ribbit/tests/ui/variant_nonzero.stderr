error: Non-zero enum requires either all non-zero discriminants *or* variant with discriminant 0 to be non-zero
 --> tests/ui/variant_nonzero.rs:6:5
  |
6 |     #[ribbit(size = 8)]
  |     ^
